README.md

```diff
@@ -1 +1,142 @@
-# hello-react-front-end
\ No newline at end of file
+<a name="readme-top"></a>
+
+<!-- TABLE OF CONTENTS -->
+
+# 📗 Table of Contents
+
+- [📖 Hello React frontend](#about-project)
+  - [🛠 Built With](#built-with)
+    - [Tech Stack](#tech-stack)
+    - [Key Features](#key-features)
+  - [🎥 Vedio Description](#vd-disc)
+- [💻 Getting Started](#getting-started)
+  - [Setup](#setup)
+  - [Prerequisites](#prerequisites)
+  - [Install](#install)
+  - [Usage](#usage)
+  - [Run tests](#run-tests)
+- [👥 Authors](#authors)
+- [🔭 Future Features](#future-features)
+- [🤝 Contributing](#contributing)
+- [⭐️ Show your support](#support)
+- [🙏 Acknowledgements](#acknowledgements)
+- [📝 License](#license)
+
+<!-- PROJECT DESCRIPTION -->
+
+# 📖 React Frontend <a name="about-project"></a>
+
+This is react front end app which uses [RAILS API](https://github.com/Daniel-Kigozi/hello-rails-back-end)for backend services
+
+## 🛠 Built With <a name="built-with"></a>
+
+### Tech Stack <a name="tech-stack"></a>
+- The app built using [Ruby on Rails](https://www.ruby-lang.org/en/).
+
+<!-- Features -->
+
+### Key Features <a name="key-features"></a>
+
+The app help you use rails as your backend application
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
+<!-- GETTING STARTED -->
+
+## 💻 Getting Started <a name="getting-started"></a>
+
+To get a local copy up and running, follow these steps...
+
+### Prerequisites
+
+In order to run this project you need:
+
+- Ruby should be installed on your local machine
+- Rails should be installed on your local machine
+- You need a Terminal.
+- You need a code editor eg: `VsCode`
+
+### Setup
+
+Clone this repository to your desired folder:
+
+Example commands:
+
+```sh
+  cd my-folder
+  git clone https://github.com/Daniel-Kigozi/hello-react-front-end.git
+```
+
+### Install
+
+Install this project with:
+
+Example command:
+
+```sh
+  cd hello-react-front-end
+  bundle install
+```
+
+### Usage
+
+To run the project, execute the following command:
+
+Example command:
+
+```sh
+  rails s
+```
+
```

Comment:
- [OPTIONAL] The correct command to run to install a react command is `npm install` and to run the project, you need to run `npm start`. I would advise you to run these commands. 👍 😎